File: backend/src/main/java/com/elbuensabor/entities/DetallePedido.java
package com.elbuensabor.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "detallePedido")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class DetallePedido extends Base {
    @Column
    private Integer cantidad;
    @Column
    private Double subtotal;

    @ManyToOne
    @JoinColumn(name = "pedido_id")
    private Pedido pedido;

    @ManyToOne
    @JoinColumn(name = "articuloManufacturado_id")
    private ArticuloManufacturado articuloManufacturado;

    @ManyToOne
    @JoinColumn(name = "articuloInsumo_id")
    private ArticuloInsumo articuloInsumo;

}

File: frontend/el-buen-sabor-frontend/src/app/entidades/DetallePedido.ts
import { ArticuloInsumo } from './ArticuloInsumo';
import { ArticuloManufacturado } from './ArticuloManufacturado';
import { Pedido } from './Pedido';

export class DetallePedido {
  id: number = 0;
  cantidad: number = 0;
  subtotal: number = 0;
  pedido: Pedido = new Pedido();
  articuloManufacturado: ArticuloManufacturado = new ArticuloManufacturado();
  articuloInsumo: ArticuloInsumo = new ArticuloInsumo();
}
